### ISSUE TITLE
Exception thrown when calling Track

### ISSUE THREAD

**Issue by gerbil802** [toothrot/amplitude-api#36]
Hi,

I'm getting an exception when calling `AmplitudeAPI.track`. Here's what I got:

```
NoMethodError: undefined method `to_h' for []:Array
/Users/abcd/.rvm/gems/ruby-2.0.0-p648/bundler/gems/amplitude-api-40903ce133aa/lib/amplitude_api/event.rb:48:in `optional_properties'
/Users/abcd/.rvm/gems/ruby-2.0.0-p648/bundler/gems/amplitude-api-40903ce133aa/lib/amplitude_api/event.rb:39:in `to_hash'
/Users/abcd/.rvm/gems/ruby-2.0.0-p648/bundler/gems/amplitude-api-40903ce133aa/lib/amplitude_api.rb:69:in `map'
/Users/abcd/.rvm/gems/ruby-2.0.0-p648/bundler/gems/amplitude-api-40903ce133aa/lib/amplitude_api.rb:69:in `track_body'
/Users/abcd/.rvm/gems/ruby-2.0.0-p648/bundler/gems/amplitude-api-40903ce133aa/lib/amplitude_api.rb:87:in `track'
```

This is what I did:
```
	  		event = AmplitudeAPI::Event.new({
		  	      user_id: "1234",
			      event_type: "Eat",
			      event_properties: {
			        id: 5031,
			        name: "Burger",
			      }
			    })
		    AmplitudeAPI.track(event)
```

**Comment by kolorahl** [comment]
Interesting, this may be a Ruby issue. What concerns me is that you seem to be on major version 2, which is what all the tests run against, but minor version 0 (zero), which I don't think anyone has tested against. Checking on my own 2.4 version, `[].to_h` results in `{}`; i.e. an empty array can be converted to an empty hash. It looks like version 2.0 may not have the functionality or may be missing the `to_h` function altogether for the Array class.

@gerbil802 Is it possible to upgrade your Ruby installation to a more recent version?

**Comment by gerbil802** [comment]
@kolorahl Solved! I upgraded to Ruby 2.1 (which is the first that contains `to_h`) and it works!

**Comment by kolorahl** [comment]
@gerbil802 Wonderful, I'm glad that worked for you!
